core/domain/summary_services.py

```diff
@@ -254,15 +254,17 @@ def get_learner_collection_dict_by_id(
         next_exploration_id = collection.first_exploration_id
         completed_exp_ids = []
 
-    # Here, the return type of 'to_dict' method is CollectionDict but for
-    # implementation purpose we are assigning LearnerCollectionDict which
-    # is inherited from CollectionDict. So, due to the difference in types
-    # MyPY throws an error. Thus to avoid the error, we used ignore here.
+    # Here we use MyPy ignore because the return type of 'to_dict' method
+    # is CollectionDict but for implementation purpose we are assigning
+    # LearnerCollectionDict which is inherited from CollectionDict. So,
```

Comment:
Shouldn't this cast rather than ignore (at the point where we specialize to LearnerCollectionDict)?

Reply:
Done cast is used!

Reply:
Updated!